test/powershell/Modules/Microsoft.PowerShell.Utility/Get-Random.Tests.ps1

```diff
@@ -163,6 +178,7 @@ Describe "Get-Random" -Tags "CI" {
         $secondRandomNumber = Get-Random 34359738367 -SetSeed 20
         $firstRandomNumber | Should -Be @secondRandomNumber
     }
+
     It "Should throw an error because the hexadecimal number is to large " {
         { Get-Random 0x07FFFFFFFFFFFFFFFF } | Should -Throw "Value was either too large or too small for a UInt32"
```

Comment:
I haven't dug deeper, but the full error message is self-contradictory (Int32 vs. UInt32): `Cannot convert value "0x07FFFFFFFFFFFFFFFF" to type "System.Int32". Error: "Value was either too large or too small for a UInt32."` Where does the *UInt* come from?

Reply:
Oh, I actually fixed this back when I was toying with the tokenizer, and then it was requested I _re-break_ this for consistency with prior implementations. This can be fixed, but the fix would be in the tokenizer and how hex numbers get parsed.

This number is an extra digit beyond the widest permitted for uint64, and a hex number with a leading zero is forced to be an unsigned numeral when using the conversion paths we have in the tokenizer (which are still done with `Convert.ToInt32()` I _think_?). I think the default path goes to (u)int32 because it checks for (u)int64 lengths and then defaults to (u)int32 for all other cases iirc.

I remember I originally did some fun things in my tokenizer PRs where I would automatically upcast between uint/int64 if needed, but it differed too much from the previous implementation I think and we decided to keep the behaviour consistent. I think my solution was to have it upcast up to decimal (or even biginteger) if needed, which worked well.

I think due to the refactoring in #7993 this error message might actually change, but it's been a little while and I don't quite recall.

Reply:
I see: if the pre-parameter-binding typing can't convert a "number-looking" argument to an actual number, it passes a string, and line `baseObject = System.Management.Automation.Language.Parser.ScanNumber((string)baseObject, typeof(int));` then re-attempts conversion, which fails.

Again, not in the scope of this PR, but I wonder if the error message could be made friendlier (and it's still curious that UInt*32* is referenced, given that hex. literals seem to top out at Int64).

Reply:
Oops, @vexx32  - wires crossed - reading your comment now.